cunumeric/module.py

```diff
@@ -1294,6 +1299,239 @@ def squeeze(a: ndarray, axis: Optional[NdShapeLike] = None) -> ndarray:
     return a.squeeze(axis=axis)
 
 
+def broadcast_shapes(*args: Sequence(Union[tuple(int), int])) -> tuple(int):
+    """
+
+    Broadcast the input shapes into a single shape.
+
+    Parameters
+    ----------
+    `*args` : tuples of ints, or ints
+        The shapes to be broadcast against each other.
+
+    Returns
+    -------
+    tuple : Broadcasted shape.
+
+    See Also
+    --------
+    numpy.broadcast_shapes
+
+    Availability
+    --------
+    Single CPU
+
+    """
+    return _broadcast_shapes(*args)
+
+
+def _broadcast_to(
+    arr: ndarray,
+    shape: Union[tuple(int), int],
+    subok: bool = False,
+    writeable: bool = False,
+    broadcasted: bool = False,
+) -> ndarray:
+    # create an array object w/ options passed from 'broadcast' routines
+    arr = array(arr, copy=False, subok=subok)
+    # 'broadcast_to' returns a read-only view of the original array
+    out_shape = broadcast_shapes(arr.shape, shape)
+    result = ndarray(
+        shape=out_shape,
+        thunk=arr._thunk.broadcast_to(out_shape),
+        broadcasted=broadcasted,
+    )
+    if result.flags["WRITEABLE"] != writeable:
+        result.setflags(write=writeable)
+    return result
+
+
+@add_boilerplate("arr")
+def broadcast_to(
+    arr: Any, shape: Union[tuple(int), int], subok: bool = False
```

Comment:
I believe `Any` should be replaced with `ndarray` here since you have added `add_boilerplate` (it will automatically convert arr to ndarray)

Reply:
Thanks for finding this out. Changed to `ndarray`